samcli/commands/build/command.py

```diff
@@ -295,6 +296,8 @@ def do_cli(  # pylint: disable=too-many-locals, too-many-statements
         build_images=processed_build_images,
         aws_region=click_ctx.region,
     ) as ctx:
+        SamApiProvider.check_implicit_api_resource_ids(ctx.stacks)
```

Comment:
Does it check whether the template is a SAM template? i.e. with `Transform: AWS::Serverless-2016-10-31`?

I believe SAM CLI also supports building pure CFN template. In that case, the implicit API logical IDs should not be reserved.

Reply:
addressed